amp-pwa-reader/src/js/Article.js

```diff
@@ -41,6 +42,61 @@ class Article {
 
   }
 
+// see https://github.com/ampproject/amphtml/blob/master/spec/amp-shadow-doc.md
+// and https://jakearchibald.com/2016/fun-hacks-faster-content/
+// Recursively read and write chunks of data from the streaming API
+  stream() {
```

Comment:
Let me know if you think that stream() is clear enough - as this is the core of what we're asking people to do.

I think I've helped by removing the recursive helper function by using `await`.  But arguably I should be farming out some of the internal logic into other methods.